src/main/java/com/easytestit/generatexml/configuration/ConfigureXMLReport.java

```diff
@@ -1,64 +1,114 @@
+ * 2 parameters when you create object of {@link ConfigureXMLReport()} class.
+ *
+ * <p>
+ *     {@code
+ *     ConfigureXMLReport conf = new ConfigureXMLReport("out/xml-report/", "TestProject");
+ *     }
+ * </p>
+ * Where:
+ * -- the first parameter is a path to JSON files that will be converted
+ * to Java object and will be aggregated to one XML file.
+ * -- the second parameter is the name of the project, an aggregated file
+ * will be created under this name. Also, if you will use functionality of sending report
+ * to report server then the report itself will be displayed under this name.
+ *  </p>
+ *
+ * <p>In additional, if you want to use functionality of sending report you only need pass
+ * 2 additional parameters in constructor of {@link ConfigureXMLReport()} class.
+ *
+ * <p>
+ *     {@code
+ *     ConfigureXMLReport conf = new ConfigureXMLReport("out/xml-report/", "TestProject", true, true);
+ *     }
+ * </p>
+ * Where additional boolean parameters are:
+ * -- the first parameter turn on functionality for creating a zip archive,
+ * which is subsequently necessary for sending to the server report.
+ * -- the second parameter turn on the functionality to send the previously created zip file
+ * to the report server.
+ * </p>
  */
 public class ConfigureXMLReport {
 
-    private static final Logger LOGGER = LogManager.getLogger(ConfigureXMLReport.class.getName());
-    private File reportFolder;
-    private Collection<String> jsonFiles;
-    private Collection<ConfigureXMLMode> configureXMLMode = new ArrayList<>();
-
-    public ConfigureXMLReport(@NotNull File reportFolder) {
-        LOGGER.info(String.format("Create configuration for application with defined '%s' report folder", reportFolder.getAbsolutePath()));
-        this.reportFolder = reportFolder;
-    }
+    private final String projectName;
+    private final File source;
+    private final boolean reportAsZip;
+    private final boolean sendReport;
 
     /**
-     * Method which added JSON files to new list
-     * @param jsonFiles the list which stored all JSON files which after should be converted to XML
+     * Configuration by default (no zip, no report)
+     *
+     * @param path path to source folder
+     * @param projectName project name
      */
-    public void addJsonFiles(Collection<String> jsonFiles) {
-        this.jsonFiles = jsonFiles;
+    public ConfigureXMLReport(final String path, final String projectName) {
+        if (path != null) {
+            this.source = new File(path);
+        } else {
+            throw new GenerateXMLReportException("Argument path should not be null but is null. See detailed stack trace: ", new NullPointerException());
+        }
+        if (projectName != null) {
+            this.projectName = projectName.replaceAll("\\s+", "");
+        } else {
+            throw new GenerateXMLReportException("Argument projectName should not be null but is null. See detailed stack trace: ", new NullPointerException());
```

Comment:
Do you really need to create an instance of  GenerateXMLReportException with 2 arguments? Perhaps it will be enough to use a constructor with 1 String argument?

Reply:
Yeah, you're right. I will make it with only one parameter.